docs/ha-considerations.md

```diff
@@ -255,60 +255,64 @@ With the services up, now the Kubernetes cluster can be bootstrapped using `kube
 
 ## kube-vip
 
-As an alternative to the more "traditional" approach of `keepalived` and `haproxy`, [kube-vip](https://kube-vip.io/) implements both management of a virtual IP and load balancing in one service. Similar to option 2 above, `kube-vip` will be run as a static pod on the control plane nodes.
+As an alternative to the more "traditional" approach of `keepalived` and `haproxy`, [kube-vip](https://kube-vip.io/) implements both management of a virtual IP and load balancing in one service. It can be implemented in bother layer2 (with ARP, and `leaderElection`) or layer3 utilising BGP peering. Similar to option 2 above, `kube-vip` will be run as a static pod on the control plane nodes.
 
 Like with `keepalived`, the hosts negotiating a virtual IP need to be in the same IP subnet. Similarly, like with `haproxy`, stream-based load-balancing allows TLS termination to be handled by the API Server instances behind it.
 
-The configuration file `/etc/kube-vip/config.yaml` looks like this:
-```yaml
-localPeer:
-  id: ${ID}
-  address: ${IPADDR}
-  port: 10000
-remotePeers:
-- id: ${PEER1_ID}
-  address: ${PEER1_IPADDR}
-  port: 10000
-# [...]
-vip: ${APISERVER_VIP}
-gratuitousARP: true
-singleNode: false
-startAsLeader: ${IS_LEADER}
-interface: ${INTERFACE}
-loadBalancers:
-- name: API Server Load Balancer
-  type: tcp
-  port: ${APISERVER_DEST_PORT}
-  bindToVip: false
-  backends:
-  - port: ${APISERVER_SRC_PORT}
-    address: ${HOST1_ADDRESS}
-  # [...]
+**NOTE**
+`kube-vip` requires access to the API server, especially during a cluster initialisation (during the `kubeadm init` phase). At this point the `admin.conf` is the only kubeconfig that is available to `kube-vip` to authenticate and communicate with the API-server. Post cluster stand up it is recommended that a user sign a custom client kubeconfig and rotate it manually on expiration.
```

Comment:
this should suffice, thanks

Reply:
>  is passed so that the go-client can speak to the API server during the kubeadm init and perform a leaderElection and bring up the HA virtual IP address.

what RBAC is needed by kube-vip exactly - i.e. what API calls does it make?
based on that we can include a minimal ClusterRole(Binding):

```
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: kube-vip
rules:
- apiGroups:
  - coordination.k8s.io
  resources:
  - leases
  verbs:
  - get
  - list
  - watch
  - create
  - update
  - patch
  - delete
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: kube-vip
subjects:
- kind: User
  name: kube-vip
  apiGroup: rbac.authorization.k8s.io
roleRef:
  kind: ClusterRole
  name: kube-vip
  apiGroup: rbac.authorization.k8s.io
```

and mention this example command:
```
kubeadm kubeconfig user --config=kubeadminit.conf --client-name=kube-vip
```

/approve
/tide merge-method-squash

Reply:
> /tide merge-method-squash

doesn't work for some reason, so please keep the commits squashed to 1.

Reply:
There are a few (mainly all required for ARP or services):

https://kube-vip.io/manifests/rbac.yaml
largely contains all of the requirements when it runs as a daemonset

Reply:
Although without the `admin.conf` being bound mounted into the static pod how will a client-go client auth/interact with the API server?

Reply:
It still needs client credentials but if i understand the problem correctly the permissions need to be pre baked. I.e. rbac rules need to exist at api server start time.

Using admin.conf entirely bypasses rbac because it signed for the system:masters group.